bpf/lib/nodeport.h

```diff
@@ -482,10 +571,18 @@ int tail_nodeport_ipv6_dsr(struct __ctx_buff *ctx)
 			    ctx_load_meta(ctx, CB_HINT), &ohead);
 #elif DSR_ENCAP_MODE == DSR_ENCAP_NONE
 	ret = dsr_set_ext6(ctx, ip6, &addr, port, &ohead);
+#elif DSR_ENCAP_MODE == DSR_ENCAP_GENEVE
+	ret = encap_geneve_dsr_opt6(ctx, ip6, &addr, port, &ifindex, &ohead);
+	if (!IS_ERR(ret)) {
+		if (ret == CTX_ACT_REDIRECT) {
+			cilium_capture_out(ctx);
+			return ctx_redirect(ctx, ifindex, 0);
+		}
+	}
```

Comment:
Maybe add a comment here why the code isn't looking the same as for IPv4 (the XDP-to-TC punt isn't needed).